pkg/controller/internal/transportadapter/transportadapter_translator.go

```diff
@@ -199,7 +201,43 @@ func (t *transportAdapterTranslator) translateTransportAdapterDeployment(
 		},
 	}
 
-	return deployment, controllerutil.SetOwnerReference(server, deployment, t.scheme)
+	// Set owner reference first
+	if err := controllerutil.SetOwnerReference(server, deployment, t.scheme); err != nil {
+		return nil, err
+	}
+
+	// Add hash annotation based on MCPServer spec to initiate a restart on changes to the MCPServer spec
+	t.addMCPServerSpecHashAnnotation(deployment, server)
+
+	return deployment, nil
+}
+
+// addMCPServerSpecHashAnnotation adds a hash annotation to the deployment's pod template
+// based on the MCPServer spec. This ensures pod restarts when the MCPServer configuration changes.
+func (t *transportAdapterTranslator) addMCPServerSpecHashAnnotation(deployment *appsv1.Deployment, server *v1alpha1.MCPServer) {
+	mcpServerSpecHash := t.computeMCPServerSpecHash(server)
+	// Add the hash to the pod template annotations
+	if deployment.Spec.Template.Annotations == nil {
+		deployment.Spec.Template.Annotations = make(map[string]string)
+	}
+	deployment.Spec.Template.Annotations["kmcp.kagent.dev/mcpserver-hash"] = mcpServerSpecHash
+}
+
+// computeMCPServerSpecHash computes a hash of the MCPServer spec
+func (t *transportAdapterTranslator) computeMCPServerSpecHash(server *v1alpha1.MCPServer) string {
+	// Hash only the fields that affect the deployment configuration for efficiency
+	hashInput := fmt.Sprintf("%s|%s|%v|%v|%v|%d|%s",
```

Comment:
Rather than creating this, can you just hash the `configYaml` directly? That way we'll never miss any newly added fields etc.

Reply:
yup, updated to has the config yaml instead.